<!doctype html>
<html>
<head>
  <meta charset="utf-8">
  <meta name="author" content="Fritz Lekschas">
  <meta name="description" content="A tool for interactive visual pattern search in sequential data">
  <meta name="keywords" content="Visual pattern search, pattern matching, sequential data, time series, epigenomics, convolutional autoencoder, unsupervised representation learning">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">

  <!-- Schema.org markup for Google+ -->
  <meta itemprop="name" content="Peax">
  <meta itemprop="description" content="A tool for interactive visual pattern search in sequential data">
  <meta itemprop="image" content="http://peax.lekschas.de/images/teaser.png">

  <!-- Twitter Card data -->
  <meta name="twitter:card" content="summary_large_image">
  <meta name="twitter:site" content="@flekschas">
  <meta name="twitter:title" content="Peax">
  <meta name="twitter:description" content="A tool for interactive visual pattern search in sequential data">
  <meta name="twitter:creator" content="@flekschas">
  <meta name="twitter:image:src" content="http://peax.lekschas.de/images/teaser.png">

  <!-- Open Graph -->
  <meta property="og:title" content="Peax" />
  <meta property="og:description" content="A tool for interactive visual pattern search in sequential data" />
  <meta property="og:type" content="website" />
  <meta property="og:url" content="https://github.com/Novartis/peax" />
  <meta property="og:image" content="http://peax.lekschas.de/images/teaser.png" />

  <title>Peax</title>

  <link rel="apple-touch-icon" href="images/touchicon.png">
  <link rel="icon" href="images/favicon.png">
</head>

<body>
  <div id="logo">
    <div id="logo-track">
      <svg></svg>
    </div>
    <svg id="logo-magnifier">
      <use
        xmlns:xlink="http://www.w3.org/1999/xlink"
        xlink:href="images/icons.svg#magnifier"></use>
    </svg>
    <div class="content">
      <div id="logo-name"><span>Peax</span></div>
    </div>
  </div>
  <div class="content-wrapper">
    <div class="content p-t-2 p-b-2">
      <!-------------------------------- TEASER ------------------------------->
      <h1>Interactive visual pattern search in sequential data using unsupervised deep representation learning</h1>
      <figure id="fig-teaser"></figure>
      <p class="abstract larger m-b-2">
        We present Peax, a novel feature-based technique for interactive visual pattern search in sequential data, like time series or data mapped to a genome sequence.
        Visually searching for patterns by similarity is often challenging because of the large search space, the visual complexity of patterns, and the user's perception of similarity.
        For example, in genomics, researchers try to link patterns in multivariate sequential data to fundamental cellular or pathogenic processes, but a lack of ground truth and high variance makes automatic pattern detection unreliable.
        We have developed a convolutional autoencoder for unsupervised representation learning of regions in sequential data that can capture more visual details of complex patterns compared to existing similarity measures.
        Using this learned representation as features of the sequential data, our accompanying visual query system enables interactive feedback-driven adjustments of the pattern search to adapt to the users' perceived similarity.
        While users label regions as either matching their search target or not, a random forest classifier learns to weigh the importance of different dimensions of the learned representation.
        We employ an active learning sampling strategy to focus the labeling process on regions that will improve the classifier in subsequent training.
        Peax is open-source, customizable, and we demonstrate its features through an extensive case study in genomics. Moreover, we report on the ability of our autoencoder models to capture visual features and evaluate the effectiveness of the learned representation for similarity search in two user studies. We show that our models retrieve significantly more similar patterns than other commonly used techniques.
      </p>

      <!------------------------------ SCREENCAST ----------------------------->
      <h3 id="screencast" class="iconized underlined anchored">
        <a href="#screencast" class="hidden-anchor">
          <svg class="icon">
            <use
              xmlns:xlink="http://www.w3.org/1999/xlink"
              xlink:href="images/icons.svg#link"></use>
          </svg>
        </a>
        <svg class="icon">
          <use
            xmlns:xlink="http://www.w3.org/1999/xlink"
            xlink:href="images/icons.svg#video"></use>
        </svg>
        <span>Screencast</span>
      </h3>
      <div id="youtube" class="video">
        <iframe
          src="https://www.youtube.com/embed/WpbBoW2f4iM?rel=0&amp;showinfo=0"
          frameborder="0"
          allowfullscreen>
        </iframe>
      </div>

      <!-------------------------------- INTRO -------------------------------->
      <h3 id="introduction" class="iconized underlined anchored">
        <a href="#introduction" class="hidden-anchor">
          <svg class="icon">
            <use
              xmlns:xlink="http://www.w3.org/1999/xlink"
              xlink:href="images/icons.svg#link"></use>
          </svg>
        </a>
        <svg class="icon icon-small">
          <use
            xmlns:xlink="http://www.w3.org/1999/xlink"
            xlink:href="images/icons.svg#text"></use>
        </svg>
        <span>Introduction</span>
      </h3>

      <section class="text">
        <figure id="fig-user-roles">
          <figcaption>Fig. 1: Exploration behavior of different user roles.</figcaption>
        </figure>

        <p class="abstract first">
          <strong>Why?</strong>  Biomedical repositories are growing rapidly and provide scientists with tremendous opportunities to re-use data. In order to exploit published data sets efficiently, it is crucial to understand the content of repositories and to discover data relevant to a question of interest. These are challenging tasks, as most repositories currently only support finding data sets through text-based search of metadata and in some cases also through metadata-based browsing. To address this, we conducted a task analysis through semi-structured interviews with 8 PhD-level domain experts and identified 3 distinct user roles (<a href="#fig-user-roles">Fig. 1</a>).
        </p>

        <figure id="fig-overview">
          <figcaption>Fig. 2: <abbr title="Semantic Annotation Tool and Ontological Relations Interface">SATORI</abbr>'s system and data model.</figcaption>
        </figure>

        <p class="abstract"><strong>What?</strong>  Biological data sets consists of experimental data and metadata describing the studies, properties of the analyzed biological samples, and attributes of individual data files. In this context, a data set is a collection of data files, along with the metadata  (<a href="#fig-overview">Fig. 2</a>). Additionally, metadata is partially annotated with ontology terms. An ontology describes a certain domain (e.g. <em>human anatomy</em>), defines controlled vocabularies for its concepts and relationships (e.g., <em>kidney</em> and <em>is-part-of</em>) and relates concepts with each other (e.g., <em>nephron is-part-of kidney</em>). By means of ontology terms, sets of annotated data sets can be classified hierarchically. <abbr title="Semantic Annotation Tool and Ontological Relations Interface">SATORI</abbr> extracts free-text and ontologically annotated metadata  (<a href="#fig-overview">Fig. 2</a>). The free-text metadata is indexed in a text-based search system. Additionally, data set-related ontology classes are parsed and visualized to provide semantic context to data sets. Since <abbr title="Semantic Annotation Tool and Ontological Relations Interface">SATORI</abbr>'s goal is to support exploration rather than to visualize ontologies themselves, only a relevant subtree of the ontologies is shown, i.e., effectively enforcing a strict containment hierarchy (<a href="#fig-ontology">Fig. 3</a>).</p>

        <div class="flex-c m-t-2 m-b-2">
          <figure id="fig-ontology" class="flex-g-1">
            <figcaption>Fig. 3: Ontology pruning.</figcaption>
          </figure>

          <figure id="fig-precision-recall" class="flex-g-1">
            <figcaption>Fig. 4: Precision and recall.</figcaption>
          </figure>
        </div>

        <figure id="fig-link-indication">
          <figcaption>Fig. 5: Link location.</figcaption>
        </figure>

        <p class="abstract"><strong>How?</strong>  <abbr title="Semantic Annotation Tool and Ontological Relations Interface">SATORI</abbr> is composed of two main interlinked views: the data set view and the exploration view. In the treemap an ontology term is illustrated by a rectangle. The area of the rectangle visualizes the size of the term relative to its sibling terms and the color indicates the distance to the farthest child term. The farther away this child term is, the darker is the color. The node-link diagram represents ontology terms as nodes and links shown parent and child terms (<a href="#fig-link-indication">Fig. 5</a>). Additionally, the diagram visualizes the precision and recall (<a href="#fig-precision-recall">Fig. 4</a>) for each term given the currently retrieved data sets. In this context, precision is useful to understand how frequently a term is used for annotation in the retrieved set of data sets and recall provides a notion of information scent by indicating if there are more data sets annotated with this term. Finally, the exploration view acts as a semantic query interface and lets users filter down collections of data sets via ontology term-based Boolean queries.</p>
      </section>

      <!----------------------------- PUBLICATION ----------------------------->
      <h3 id="publications" class="iconized underlined anchored">
        <a href="#publications" class="hidden-anchor">
          <svg class="icon">
            <use
              xmlns:xlink="http://www.w3.org/1999/xlink"
              xlink:href="images/icons.svg#link"></use>
          </svg>
        </a>
        <svg class="icon">
          <use
            xmlns:xlink="http://www.w3.org/1999/xlink"
            xlink:href="images/icons.svg#document"></use>
        </svg>
        Publication
      </h3>
      <ol id="publications" class="no-list-style">
        <li class="reference">
          <p class="title"><a href="https://doi.org/10.1093/bioinformatics/btx739" target="_blank">SATORI: A System for Ontology-Guided Visual Exploration of Biomedical Data Repositories</a></p>
          <div class="secondary-information">
            <ol class="no-list-style inline-list authors">
              <li>Fritz Lekschas</li>
              <li>Nils Gehlenborg</li>
            </ol>
            <span class="journal">Bioinformatics</span>, Volume 34, Issue 7, 1 April 2018, Pages 1200–1207.
            <span class="doi"><abbr title="Digital Object Identifier">doi</abbr>: <a href="https://doi.org/10.1093/bioinformatics/btx739" target="_blank">10.1093/bioinformatics/btx739</a></span>
          </div>
        </li>
      </ol>

      <!----------------------------- SOURCE CODE ----------------------------->
      <h3 id="source-code" class="iconized underlined anchored">
        <a href="#source-code" class="hidden-anchor">
          <svg class="icon">
            <use
              xmlns:xlink="http://www.w3.org/1999/xlink"
              xlink:href="images/icons.svg#link"></use>
          </svg>
        </a>
        <svg class="icon">
          <use
            xmlns:xlink="http://www.w3.org/1999/xlink"
            xlink:href="images/icons.svg#code"></use>
        </svg>
        Source Code
      </h3>
      <p>All the code of HiPiler are publicly accessible and open-source.</p>
      <ul class="no-list-style large-spacing iconized smaller">
        <li class="iconized">
          <svg class="icon">
            <use
              xmlns:xlink="http://www.w3.org/1999/xlink"
              xlink:href="images/icons.svg#github"></use>
          </svg><a href="https://github.com/flekschas/hipiler" target="_blank">https://github.com/flekschas/hipiler</a>
        </li>
      </ul>

      <!------------------------------- AUTHORS ------------------------------->
      <h3 id="authors" class="iconized underlined anchored">
        <a href="#authors" class="hidden-anchor">
          <svg class="icon">
            <use
              xmlns:xlink="http://www.w3.org/1999/xlink"
              xlink:href="images/icons.svg#link"></use>
          </svg>
        </a>
        <svg class="icon">
          <use
            xmlns:xlink="http://www.w3.org/1999/xlink"
            xlink:href="images/icons.svg#people"></use>
        </svg>
        Authors
      </h3>
      <ol id="authors-list" class="no-list-style">
        <li>
          <h4 class="blue">Fritz Lekschas</h4>
          <p class="smaller">Harvard John A. Paulson School of Engineering and Applied Sciences</p>
          <ul class="no-list-style inline-list iconized remix-links">
            <li>
              <svg class="icon">
                <use
                  xmlns:xlink="http://www.w3.org/1999/xlink"
                  xlink:href="images/icons.svg#globe"></use>
              </svg>
              <a href="https://lekschas.de" target="_blank">https://lekschas.de</a>
            </li>
            <li>
              <svg class="icon">
                <use
                  xmlns:xlink="http://www.w3.org/1999/xlink"
                  xlink:href="images/icons.svg#twitter"></use>
              </svg>
              <a href="https://twitter.com/flekschas" target="_blank">flekschas</a>
            </li>
            <li>
              <svg class="icon">
                <use
                  xmlns:xlink="http://www.w3.org/1999/xlink"
                  xlink:href="images/icons.svg#github"></use>
              </svg>
              <a href="https://github.com/flekschas" target="_blank">flekschas</a>
            </li>
          </ul>
        </li>
        <li>
          <h4 class="blue">Nils Gehlenborg</h4>
          <p class="smaller">Department of Biomedical Informatics, Harvard Medical School</p>
          <ul class="no-list-style inline-list iconized remix-links">
            <li>
              <svg class="icon">
                <use
                  xmlns:xlink="http://www.w3.org/1999/xlink"
                  xlink:href="images/icons.svg#globe"></use>
              </svg>
              <a href="http://gehlenborglab.org" target="_blank">http://gehlenborglab.org</a>
            </li>
            <li>
              <svg class="icon">
                <use
                  xmlns:xlink="http://www.w3.org/1999/xlink"
                  xlink:href="images/icons.svg#twitter"></use>
              </svg>
              <a href="https://twitter.com/nils_gehlenborg" target="_blank">nils_gehlenborg</a>
            </li>
            <li>
              <svg class="icon">
                <use
                  xmlns:xlink="http://www.w3.org/1999/xlink"
                  xlink:href="images/icons.svg#github"></use>
              </svg>
              <a href="https://github.com/ngehlenborg" target="_blank">ngehlenborg</a>
            </li>
          </ul>
        </li>
      </ol>

      <!------------------------------ REFERENCES ----------------------------->
      <h3 id="references" class="iconized underlined anchored">
        <a href="#references" class="hidden-anchor">
          <svg class="icon">
            <use
              xmlns:xlink="http://www.w3.org/1999/xlink"
              xlink:href="images/icons.svg#link"></use>
          </svg>
        </a>
        <svg class="icon">
          <use
            xmlns:xlink="http://www.w3.org/1999/xlink"
            xlink:href="images/icons.svg#bookmark"></use>
        </svg>
        References
      </h3>
      <ol class="no-list-style large-spacing iconized">
        <li id="ref-1" class="flex-c reference">
          <span class="number">[1]</span>
          <span class="flex-g-1">Fraser and Bickmore. (2017) Nuclear organization of the genome and the potential for gene regulation. <em>Nature</em>, 447, 7143, 413–417.</span>
        </li>
      </ol>
    </div>

    <footer id="footer">
      <div class="content">
        <div class="flex-c flex-jc-c flex-a-c">
          <div class="vcg-name flex-c flex-d-c">
            <span>Visual</span>
            <span>Computing</span>
            <span>Group</span>
          </div>
          <svg class="logo-vcg">
            <use
              xmlns:xlink="http://www.w3.org/1999/xlink"
              xlink:href="images/icons.svg#harvard-vcg"></use>
          </svg>
          <svg class="logo-seas">
            <use
              xmlns:xlink="http://www.w3.org/1999/xlink"
              xlink:href="images/icons.svg#harvard-seas"></use>
          </svg>
          <svg class="logo-hms">
            <use
              xmlns:xlink="http://www.w3.org/1999/xlink"
              xlink:href="images/icons.svg#harvard-hms"></use>
          </svg>
          <svg class="logo-novartis">
            <use
              xmlns:xlink="http://www.w3.org/1999/xlink"
              xlink:href="images/icons.svg#novartis"></use>
          </svg>
        </div>
      </div>
      <div id="more-at-lekschas-de" class="flex-c flex-jc-c">
        <a href="https://lekschas.de/research">More Projects by Fritz Lekschas</a>
      </div>
    </footer>
  </div>
</body>
</html>
